pkg/libvmi/network.go

```diff
@@ -132,3 +134,10 @@ func WithSubdomain(subdomain string) Option {
 		vmi.Spec.Subdomain = subdomain
 	}
 }
+
+// WithoutAutoAttachPodNetwork explicitly requests to disable the automatic attachment of pod network
+func WithoutAutoAttachPodNetwork() Option {
```

Comment:
I think this function should not be integral part of the library, the only place that use it is unit-tests is could be defined at the tests file, please consider moving it there.

Also I find it odd having a function that says `Without...`, it limits the configuration for specific state (passing the *bool would enable other states, maybe `WithAutoAttach..Settings`).
Note that it introduce dependency on `kubevirt.io/kubevirt/pkg/pointer`, it could make it harder to consume outside this repository.

Reply:
> I think this function should not be integral part of the library, the only place that use it is unit-tests is could be defined at the tests file, please consider moving it there.

The PR description of https://github.com/kubevirt/kubevirt/pull/11516 states:
> By promoting libvmi at the root pkg/, we allow both unit and e2e tests to use its services.

There are at least three other places in the code that could benefit from having it [1][2][3].
Do you see a reason not to have it?

> Also I find it odd having a function that says `Without...`, it limits the configuration for specific state (passing the *bool would enable other states, maybe `WithAutoAttach..Settings`). Note that it introduce dependency on `kubevirt.io/kubevirt/pkg/pointer`, it could make it harder to consume outside this repository.

The logic behind this field behaves the same if the field is set to `nil` or to a pointer to `true`.
So the other interesting scenario is testing the logic when the field is set to a pointer to `false`.

libvmi is already dependent on `kubevirt.io/kubevirt/pkg/pointer`, please see [4] and [5].

[1] https://github.com/kubevirt/kubevirt/blob/6d26a21376f13149452e1394bd46820ed90b0a27/tests/network/vmi_networking.go#L421
[2] https://github.com/kubevirt/kubevirt/blob/6d26a21376f13149452e1394bd46820ed90b0a27/tests/network/vmi_networking.go#L440
[3] https://github.com/kubevirt/kubevirt/blob/aa2d49325e6fb4afe9814f15f94d6ca4e83e8afd/tests/security_features_test.go#L94
[4] https://github.com/kubevirt/kubevirt/blob/23eaf8d0d0a3cc5b762c9d90e8abb44062a44aac/pkg/libvmi/BUILD.bazel#L19
[5] https://github.com/kubevirt/kubevirt/blob/810ad06c3fa47d7a9cefb6040aee2870a1552d34/pkg/libvmi/vmi.go#L30

Reply:
Furthermore, a function with the `without` prefix already exists: https://github.com/kubevirt/kubevirt/blob/810ad06c3fa47d7a9cefb6040aee2870a1552d34/pkg/libvmi/vmi.go#L259

Reply:
The way auto-attach network API works seem different than most our API examples and somewhat not trivial, from what I understand, it's set to be true by default unless otherwise is configured.

I would still avoid defining this function to have "without" style and change it to receive the actual desired field state (*bool in this case).
Although there are some examples for such functions, I think it make more sense having function to describe how the desired object will be instead of describing what it shouldn't be.
In addition, hardcoing the auto-attach setting to false and  function name saying `Without..` is somewhat exposes some details about how the API works, such details libvmi should not be aware off IMO, consider the following example:
```
func WithoutAutoAttachPodNetwork(autoAttach *bool) Option {
	return func(vmi *kvirtv1.VirtualMachineInstance) {
   		vmi.Spec.Domain.Devices.AutoattachPodInterface = autoAttach
...
```

Reply:
Done.